src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+import BN from 'bignumber.js'
+import { Hop } from '@hop-protocol/sdk'
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import { v4 as uuidv4 } from 'uuid'
+import { createClient } from 'urql'
+import { Wallet } from 'ethers'
+import { SwapProvider } from '../SwapProvider'
+import cryptoassets from '../../utils/cryptoassets'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils'
+import { isERC20 } from '../../utils/asset'
+import {
+  BaseSwapProviderConfig,
+  EstimateFeeRequest,
+  EstimateFeeResponse,
+  NextSwapActionRequest,
+  QuoteRequest,
+  SwapRequest,
+  SwapStatus,
+} from '../types';
+import { ActionContext } from '../../store'
+import { Network} from '../../store/types'
+
+export interface HopSwapProviderConfig extends BaseSwapProviderConfig {
+  graphqlBaseURL: string;
+}
+
+class HopSwapProvider extends SwapProvider {
+  config: HopSwapProviderConfig;
+  graphqlURLs: { [key: string]: string };
+
+  constructor(config: HopSwapProviderConfig) {
+    super(config)
+    this.graphqlURLs = {
+      url: this.config.graphqlBaseURL,
+      ethereum: 'hop-mainnet',
+      xdai: 'hop-xdai',
+      arbitrum: 'hop-arbitrum',
+      polygon: 'hop-polygon',
+      optimism: 'hop-optimism'
+    }
+  }
+  /**
+   * Get the supported pairs of this provider for this network
+   * @param {{ network }} network
+   */
+  // eslint-disable-next-line no-unused-vars
+  async getSupportedPairs() {
+    return [];
+  }
+
+  confirmationsSendCount(networkName: string) {
+      const networkToConfirmations: { [key: string]: number } =    
+      {
+        ethereum: 1,
+        arbitrum: 20,
+        polygon: 256
+      }
+      return networkToConfirmations[networkName]
+  }
+
+  gasLimit(networkName: string){
+    const networkToGasLimit : { [key: string]: any } = {
+      arbitrum: {
```

Comment:
Why hardcoded? we cannot estimate these using rpc?

Reply:
It is done like this because 
- when you want to call estimate fee on the send method it throws an error because it cannot be estimated, because the approve is still not done (maybe the problem was appearing on Arbitrum afaik)
- their team suggested these values, so I used them

The problem is similar to  this one, tho 
https://github.com/liquality/wallet-core/blob/master/src/swaps/sovryn/SovrynSwapProvider.ts#L296